### DISCUSSION TITLE
Milvus Standalone主从复制部署

### DISCUSSION THREAD

**Discussion by wxd1120** [milvus-io/milvus#38337] [Q&A and General discussion]
你好，我看到官方文档中说，Milvus Standalone 通过主从复制支持高可用性，但是没看到有具体的standalone主从部署的文档，这个有相关文档吗

**Comment by yhmo** [comment]
see see这里：https://milvus.io/docs/coordinator_ha.md#Coordinator-HA

**Reply by wxd1120** [reply]
您好，谢谢回复！
但我看您给的文档是集群模式下协调器HA的方案，我看standalone实际部署下来其实只有三个container：milvus-standalone、 milvus-etcd、milvus-minio，standalone模式下，root coordinator并不是一个单独的容器，而是包含在milvus-standalone容器内，和其他协调器、work node共享一个容器的。

我们数据量不大，所以不想用集群模式，但是单机版的standalone模式又存在单点问题，所以想了解下standalone是否有HA的解决方案

我的想法是，如果要在standalone模式下保持HA，是不是得etcd、和minio都得用外置的集群模式，然后milvus-standalone采用多个副本，配置上是不是可以借鉴https://milvus.io/docs/coordinator_ha.md#Coordinator-HA这个来配置？

**Reply by yhmo** [reply]
standalone就不能用Coordinator HA了，那个是cluster用的。
standalone有两种：
- 一种是使用外置etcd/minio的：https://milvus.io/docs/install_standalone-docker-compose.md  这种有三个容器，如果要搞多个standalone，就得给每个standalone都配单独的etcd和minio，如果共用etcd，那么etcd挂了就都完了。共用etcd/minio，做法上是可以，只要把rootPath区分开，只不过不HA。
- 另一种是内置etcd，使用本地磁盘直接读写： https://milvus.io/docs/install_standalone-docker.md 这种仅会起一个容器，所有东西都包含在这个容器里，那么主从就只要维护两个容器即可。

然后怎么保证两个standalone里的数据一致是个问题(你的业务同时向两个实例写入数据)，这个我也没有什么idea，粗糙点就失败了retry好了。

**Reply by xujingnan308** [reply]
> standalone就没不能用Coordinator HA了，那个是cluster用的。 standalone有两种：
> 
> * 一种是使用外置etcd/minio的：https://milvus.io/docs/install_standalone-docker-compose.md  这种有三个容器，如果要搞多个standalone，就得给每个standalone都配单独的etcd和minio，如果共用etcd，那么etcd挂了就都完了。共用etcd/minio，做法上是可以，只要把rootPath区分开，只不过不HA。
> * 另一种是内置etcd，使用本地磁盘直接读写： https://milvus.io/docs/install_standalone-docker.md 这种仅会起一个容器，所有东西都包含在这个容器里，那么主从就只要维护两个容器即可。
> 
> 然后怎么保证两个standalone里的数据一致是个问题(你的业务同时向两个实例写入数据)，这个我也没有什么idea，粗糙点就失败了retry好了。

我们也有类似的问题
集群模式太重了，lite又没办法统一管理。standalone好像比较可行，本来以为有HA的，实际是还没有对吧？

**Reply by xiaofan-luan** [reply]
standalone不会提供HA模式，目前只有分布式集群才是高可用的。
Standalone 依赖分布式minio和kafka，结合K8s，也可以实现分钟级别的FO。
也许考虑一下我们的托管服务https://zilliz.com.cn/pricing?

**Reply by xujingnan308** [reply]
> standalone不会提供HA模式，目前只有分布式集群才是高可用的。 Standalone 依赖分布式minio和kafka，结合K8s，也可以实现分钟级别的FO。 也许考虑一下我们的托管服务https://zilliz.com.cn/pricing?

“Standalone 依赖分布式minio和kafka，结合K8s，也可以实现分钟级别的FO”
能详细讲讲这个方案吗？有没有相关参考文档？我们团队正在做向量数据库的选型

托管服务在一些私有云场景是不是搞不定？

**Reply by xiaofan-luan** [reply]
其实就是用helm部署，minio起4副本，etcd3副本，kafka3副本最小部署。
MIlvus可以只起一个Standalone 节点

**Reply by wxd1120** [reply]
谢谢各位大佬回复，我再看看
